projects/thales/utils.js

```diff
@@ -0,0 +1,80 @@
+const { sumTokens2 } = require('../helper/unwrapLPs')
+const abi = require('./abi.json')
+const CHAIN_CONFIG = require('./chainConfig')
+
+const utils = {
+  // Get token address from chain config
+  getTokenAddress: (chain, symbol) => CHAIN_CONFIG[chain].tokens[symbol],
+
+  // Add LP TVL
+  addLPTvl: async (api, contract, token) => {
+    const tokenAddress = utils.getTokenAddress(api.chain, token)
+    api.add(tokenAddress, await api.call({
+      target: contract,
+      abi: abi.totalDeposited
```

Comment:
where does the deposited tokens go? I dont see it on the contract

Reply:
Good question! The AMM LP tokens are deposited into round pool contracts, which are independently created for each new round. 
To verify the actual token balance against `totalDeposited`:

1. Get the current round via `round`
2. Check these contract addresses' balances:
   - Current round pool: roundPools[round]
   - Next round pool: roundPools[round + 1] (for pre-deposits)
   - Default round pool: roundPools[1] (for cross-round trades)